aten/src/ATen/native/cuda/MultiTensorApply.cuh

```diff
@@ -132,6 +132,8 @@ void multi_tensor_apply(
     if (tensor_lists[0][t].numel() == 0) {
       n_zero_tensors++;
       if (t != n_tensors - 1) {
+        // allow launching of the last kernel with accumulated
```

Comment:
what do you think of the following:
1) remove check on line 134, unconditionally cycle if there's a zero-sized tensor
2) remove block on line 150, you'll never get there
3) remove last_chunk computation on line 167 and last chunk condition on line 169
4) flush remaining blocks on line 197 if loc_block_info is non-zero, regardless of chunks

Does it work?
1) all zero tensors, we never go into meat of the loop, loc_block_info is zero, nothing is launched, good
2) a string of zero element tensors at the end, we never go into loop for them, but loc_block_info is (probably) nonzero. It it's nonzero, we'll flush whatever was accumulated, if it's already zero, we are good.

Reply:
I did have this as a fleeting thought--now that we have test cases, what you're saying sounds reasonable and wouldn't hurt to try! I'm running the tests now--will let u know how it goes

Reply:
looks promising!

Reply:
updated PR to be simpler :) thanks @ngimel for the adept suggestions